src/Simulation.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Simulation.h"
+
+void SimulationManager::RegisterSimObj(SimObj* simObj) {
+    dg_assert(simObj != nullptr, "dont give null ptr to this function euge.");
+    dg_assert(simObj->_id < MAX_SIM_OBJECTS, "too many sim objects");
+
+    size_t id = (size_t)simObj->_id;
+    spatials[id] = simObj->GetComponent<Spatial>(ComponentType::Spatial);
```

Comment:
yea i noticed that, i think the easy way around it for now is just have the simulation class run through every entity and 'remake' the arrays kinda, not sure otherwise how to handle it

Reply:
https://github.com/alecthomas/entityx/blob/master/entityx/Entity.h#L947. I hate this guys code, but this is a way. Same way I've done it. Sometime of class responsible for holding all the entities and components that the entity just has a pointer too.